test/e2e/lib/framework/serverApps/intake.ts

```diff
@@ -97,11 +118,48 @@ async function readSessionReplay(req: express.Request): Promise<SessionReplayCal
     })
 
     req.busboy.on('finish', () => {
-      segmentPromise.then((segment) => resolve({ metadata, segment })).catch((e) => reject(e))
+      segmentPromise
+        .then((segment) => {
+          events.push('sessionReplay', { metadata, segment })
+        })
+        .then(resolve)
+        .catch((e) => reject(e))
     })
   })
 }
 
+async function forwardReplayToIntake(req: express.Request): Promise<any> {
+  return new Promise((resolve, reject) => {
+    const form = new FormData()
+    req.busboy.on('field', (name, value, _info) => {
+      form.append(name, value)
+    })
+    req.busboy.on('file', (name, file, { filename }) => {
+      form.append(name, file, { filename })
+      const intakeRequest = prepareIntakeRequest(req, form.getHeaders())
+      form.pipe(intakeRequest)
+      intakeRequest.on('response', resolve)
+      intakeRequest.on('error', reject)
+    })
+  })
+}
+
+function prepareIntakeRequest(req: express.Request, headers: object) {
+  const ddforward = req.query.ddforward! as string
+  if (!/^https:\/\/(session-replay|rum|logs)\.browser-intake-datadoghq\.com\//.test(ddforward)) {
+    throw new Error(`Unsupprted ddforward: ${ddforward}`)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏Typo
```suggestion
    throw new Error(`Unsupported ddforward: ${ddforward}`)
```